demos/mask_rcnn_demo/cpp/main.cpp

```diff
@@ -45,240 +49,327 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+
+    slog::info << "Preprocessor configuration: " << slog::endl;
+    slog::info << ppp << slog::endl;
+    model = ppp.build();
+
+    // set batch size
+    ov::set_batch(model, modelBatchSize);
+    slog::info << "\tBatch size is set to " << modelBatchSize << slog::endl;
+
+    auto startTime = std::chrono::steady_clock::now();
+    for (size_t i = 0, inputIndex = 0; i < modelBatchSize; i++, inputIndex++) {
+        if (inputIndex >= imagePaths.size()) {
+            inputIndex = 0;
+        }
 
-            /** Fill first input tensor with images. First b channel, then g and r channels **/
-            if (inputInfoItem.second->getTensorDesc().getDims().size() == 4) {
-                /** Iterate over all input images **/
-                for (size_t image_id = 0; image_id < images.size(); ++image_id)
-                    matToBlob(images[image_id], input, image_id);
-            }
+        cv::Mat image = cv::imread(imagePaths[inputIndex], cv::IMREAD_COLOR);
+
+        if (image.empty()) {
+            slog::warn << "Image " + imagePaths[inputIndex] + " cannot be read!" << slog::endl;
+            continue;
+        }
 
-            /** Fill second input tensor with image info **/
-            if (inputInfoItem.second->getTensorDesc().getDims().size() == 2) {
-                InferenceEngine::LockedMemory<void> inputMapped =
-                    InferenceEngine::as<InferenceEngine::MemoryBlob>(input)->wmap();
-                auto data = inputMapped.as<float *>();
-                data[0] = static_cast<float>(netInputHeight);  // height
-                data[1] = static_cast<float>(netInputWidth);  // width
-                data[2] = 1;
+        images.push_back(image);
+    }
+    if (images.empty())
+        throw std::logic_error("Valid input images were not found!");
+
+    // Load model to the device
+    ov::CompiledModel compiled_model = core.compile_model(model, FLAGS_d);
+    logCompiledModelInfo(compiled_model, FLAGS_m, FLAGS_d);
+
+    // Create Infer Request
+    ov::InferRequest infer_request = compiled_model.create_infer_request();
+
+    // Set image input data
+    for (size_t idx = 0; idx < inputs.size(); idx++) {
+        ov::Tensor tensor = infer_request.get_input_tensor(idx);
+        ov::Shape shape = tensor.get_shape();
+
+        if (shape.size() == 4) {
+            for (size_t batchId = 0; batchId < modelBatchSize; ++batchId) {
+                if (old_model) {
+                    matToTensor(images[batchId], tensor, batchId);
+                } else {
+                    cv::Size size = {
+                        int(image_tensor_width),
+                        int(image_tensor_height)
+                    };
+                    unsigned char* data = tensor.data<unsigned char>() + batchId * image_tensor_width * image_tensor_height * 3;
+                    cv::Mat image_resized(size, CV_8UC3, data);
+                    cv::resize(images[batchId], image_resized, size);
```

Comment:
This is a job for `resize2Tensor()` :)

Reply:
`resize2tensor` does not work with batch